cowbird/permissions_synchronizer.py

```diff
@@ -199,6 +209,26 @@ def _generate_regex_from_segments(res_segments: List[ConfigSegment]) -> Tuple[st
         res_regex += r"$"
         return res_regex, named_segments_count
 
+    @staticmethod
+    def _generate_nametype_path_from_segments(res_segments: List[ConfigSegment], src_resource_tree: ResourceTree) -> str:
+        """
+        Generate nametype path (ex.: /name1::type1/name2::type2 where name can be a field found in ResourceSegment).
+
+        :param res_segments: list of segments
+        :param src_resource_tree: Resource tree associated with the permission to synchronize
+        """
+        resource_nametype_path = ""
+        index = 0
+        for res in src_resource_tree:
+            if index < len(res_segments):
+                key = res_segments[index].get("field") if res_segments[index].get("field") is not None else "resource_name"
```

Comment:
Does it mean it would be possible to have a path that includes a mix of both `resource_name`s and `resource_display_name`s? Is there a use case for this?

Reply:
Yes. That is the case for STAC pointing at THREDDS.
`/collections` is using the typical `resource_name` (route `collections` under `stac` service), 
but `/collections/{collectionID}` contains the `thredds:<path/to/dataset` in `resource_display_name` because `collectionID` can be anything that does not have a corresponding name convention with the THREDDS dataset it represents.